common/cluster/metadata.go

```diff
@@ -130,7 +130,7 @@ type (
 		versionToClusterName map[int64]string
 
 		clusterCallbackLock   sync.RWMutex
-		clusterChangeCallback map[string]CallbackFn
+		clusterChangeCallback map[any]CallbackFn
```

Comment:
`comparable`?

Reply:
it need to be comparable. not sure any would work.

Reply:
I believe `any` here is an alias for `interface{}` which is a type name and comparable. `comparable` is a constraint which doesn't work here.

Reply:
@yycptt is right. And it works here as it uses the struct pointer as key.